specification/search/resource-manager/Microsoft.Search/preview/2019-10-01-preview/search.json

```diff
@@ -1106,6 +1423,190 @@
+      "properties": {
+        "value": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkResource"
+          },
+          "description": "The list of supported Private Link Resources."
+        }
+      },
+      "description": "Response containing a list of supported Private Link Resources.",
+      "x-ms-external": true
+    },
+    "PrivateLinkResource": {
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The ID of the private link resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the private link resource."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The resource type."
+        },
+        "properties": {
+          "$ref": "#/definitions/PrivateLinkResourceProperties",
+          "readOnly": true,
+          "description": "Describes the properties of a supported private link resource for the Azure Cognitive Search service."
+        }
+      },
+      "description": "Describes a supported private link resource for the Azure Cognitive Search service."
+    },
+    "PrivateLinkResourceProperties": {
+      "properties": {
+        "groupId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The group ID of the private link resource."
+        },
+        "requiredMembers": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of required members of the private link resource."
+        },
+        "requiredZoneNames": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The list of required DNS zone names of the private link resource."
+        },
+        "shareablePrivateLinkResourceTypes": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ShareablePrivateLinkResourceType"
+          },
+          "description": "The list of resources that are onboarded to private link service, that are supported by Azure Cognitive Search."
+        }
+      },
+      "description": "Describes the properties of a supported private link resource for the Azure Cognitive Search service. For a given API version, this represents the 'supported' groupIds when creating a shared private link resources."
+    },
+    "ShareablePrivateLinkResourceType": {
+      "properties": {
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The name of the resource type that has been onboarded to private link service, supported by Azure Cognitive Search."
+        },
+        "properties": {
```

Comment:
You might want to consider x-ms-client-flatten (when nest properties inside properties)
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-client-flatten

If it is shared from other resource provider then don't change.

Reply:
I think we're just gonna leave it, this particular nested properties is only an output (i.e. customers can't set the values, can only retrieve them) and it is based on a spec from the networking team.